InformationScripting/src/query_prompt/QueryBuilder.cpp

```diff
@@ -89,7 +95,14 @@ Query* QueryBuilder::visitOperator(QueryBuilder* self, OperatorQueryNode* op)
 	else if (leftComposite)
 	{
 		// union
-		connectQueriesWith(composite, leftComposite, new UnionOperator(), right);
+		Query* unionQuery = nullptr;
+		if (op->op() == OperatorQueryNode::OperatorTypes::Pipe)
+			unionQuery = composite->addQuery(std::unique_ptr<Query>(new UnionOperator()));
```

Comment:
I don't agree, it is more descriptive the way it is, even though the representation is a | the operator performs a union.